File: examples/doctor_class/Doctor.rb
require_relative "../human_class/Human.rb"

class Doctor < Human
    def initialize(opt)
        super
        @specialty = opt["specialty"]
    end

    def saySpecialty
        puts "My specialty is " + @specialty.to_s
    end
end

if $0 == __FILE__
    doctor1 = Doctor.new({"name" => "FirstName LastName", "sex" => "male", "specialty" => "cardiology"})
    doctor1.sayName
    doctor1.saySex
    doctor1.saySpecialty
end

File: examples/class/doctor_class/Doctor.js
const Human = require('../human_class/Human.js');
class Doctor extends Human {
	constructor(opt) {
		super(opt);
        this.specialty = opt["specialty"];
    }

	saySpecialty() {
        console.log("My specialty is " + this.specialty);
    }
}
module.exports = Doctor;

if(!module.parent) {
    let doctor1 = new Doctor({"name":"FirstName LastName", "sex":"male", "specialty":"cardiology"});
    doctor1.sayName();
    doctor1.saySex();
    doctor1.saySpecialty();
}
